op-node/rollup/derive/batches.go

```diff
@@ -37,22 +38,26 @@ const (
 // conform with the constraints of the derivation pipeline. The resulting L1 origin will always be
 // the same as parent's or one block after parent's, will always conform to the derivation
 // constraints, and is deterministic given `parent` and `suggested.`
-func EspressoL1Origin(cfg *rollup.Config, parent eth.L2BlockRef, suggested eth.L1BlockRef, nextL1Block eth.L1BlockRef, l log.Logger) uint64 {
+func EspressoL1Origin(cfg *rollup.Config, parent eth.L2BlockRef, suggested eth.L1BlockRef, fetchNextL1Block func() (eth.L1BlockRef, error), l log.Logger) (uint64, error) {
 	prev := parent.L1Origin
 	windowStart := parent.Time + cfg.BlockTime
-	nextL1BlockEligible := nextL1Block.Time <= windowStart
 
 	// Constraint 1: the L1 origin must not skip an L1 block.
 	if suggested.Number > prev.Number+1 {
+		nextL1Block, err := fetchNextL1Block()
+		if err != nil {
+			return 0, fmt.Errorf("failed to fetch next possible L1 origin %d: %w", nextL1Block, err)
+		}
+		nextL1BlockEligible := nextL1Block.Time <= windowStart
```

Comment:
Nit: put this after the comment, which is explaining why it's ok to get this and we won't block. Can do this in a separate PR, though. Let's get this merged ASAP